lib/CGameInfoCallback.cpp

```diff
@@ -689,6 +689,24 @@ CGameInfoCallback::CGameInfoCallback(CGameState *GS, boost::optional<PlayerColor
 const std::vector< std::vector< std::vector<ui8> > > & CPlayerSpecificInfoCallback::getVisibilityMap() const
 {
 	//boost::shared_lock<boost::shared_mutex> lock(*gs->mx);
+	if(player == PlayerColor::SPECTATOR) // TODO: need to be replaced somehow
+	{
+		static std::vector<std::vector<std::vector<ui8> > > map;
```

Comment:
having static object does not make sense if you rebuilding it on each call, also it is not thread-safe.